src/app/application.cpp

```diff
@@ -515,21 +523,10 @@ bool Application::sendParams(const QStringList &params)
     return m_instanceManager->sendMessage(params.join(PARAMS_SEPARATOR));
 }
 
-// As program parameters, we can get paths or urls.
-// This function parse the parameters and call
-// the right addTorrent function, considering
-// the parameter type.
-void Application::processParams(const QStringList &params)
+Application::Params Application::parseParams(const QStringList &params) const
```

Comment:
> Seems this can be moved out of class.

Then I need either make Application::Params public (but it is intended for private use only) or make the function friend. I dislike both of them.

Reply:
>To name a few: `ProgramParameters`, `AppParameters`...

It is already `Application::Params` so `Program`, `App` etc. are redundant.
>`TorrentStartupParams`

IMO, `Startup` part is confusing.
Maybe `Application::AddTorrentParams`?

Reply:
>Maybe Application::AddTorrentParams?

OK.